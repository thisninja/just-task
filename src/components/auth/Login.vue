<template>
  <div class="full-width">
    Login form
  </div>
</template>

<script>
export default {
  name: 'Login',
}
</script>
