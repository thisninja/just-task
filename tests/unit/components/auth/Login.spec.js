import Vuex from 'vuex';
import VueRouter from 'vue-router';
import VueMaterial from 'vue-material'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Login from '@/components/auth/Login.vue';

const localVue = createLocalVue();

localVue.use(VueMaterial);
localVue.use(VueRouter);
localVue.use(Vuex);

describe('Login.vue', () => {
  let wrapper;
  let store;
  let actions;
  let mutations;
  let getters;

  beforeEach(() => {
    getters = getters || {
      isAuthenticated: () => true,
    };

    actions = actions || {
      login: jest.fn(),
    };

    mutations = mutations || {
      authUser: jest.fn(),
    };

    store = new Vuex.Store({
      getters,
      actions,
      mutations,
    });

    wrapper = shallowMount(Login, {
      store,
      localVue,
    });
  });

  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('onSubmit', () => {
      afterEach(() => {
        jest.clearAllMocks();
      });

      it('should trigger onSubmit', () => {
        const mock = jest.fn();
        wrapper.vm.onSubmit = mock;
        wrapper.find(".auth-form").trigger("submit.prevent")

        expect(mock).toHaveBeenCalled();
      });

      it('should trigger isValidFormByType with param "login"', () => {
        const mock = jest.fn();
        wrapper.vm.isValidFormByType = mock;
        wrapper.vm.onSubmit();

        expect(mock).toHaveBeenCalledWith('login');
        expect(actions.login).not.toHaveBeenCalled();
      });

      it('should trigger login action if isValidFormByType returned true', () => {
        const mock = jest.fn().mockImplementationOnce(() => true);
        wrapper.vm.isValidFormByType = mock;
        wrapper.vm.onSubmit();

        expect(actions.login).toHaveBeenCalled();
      });
    });
  });
});
