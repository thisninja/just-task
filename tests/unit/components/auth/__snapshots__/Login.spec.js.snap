// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue snapshots should match snapshot 1`] = `
<div class="full-width">
  <common-placeholder-stub title="Log into your account to use the app"></common-placeholder-stub>
  <!---->
  <form novalidate="novalidate" class="auth-form">
    <md-field-stub mdcounter="true" mdtogglepassword="true" class=""><label>
        Enter your email
      </label>
      <md-input-stub value="" required="true" id="md-input-uiflt5o9r" type="text"></md-input-stub>
    </md-field-stub>
    <md-field-stub mdcounter="true" mdtogglepassword="true" class=""><label>
        Enter your password
      </label>
      <md-input-stub value="" required="true" id="md-input-qyd7gti1pg" type="password"></md-input-stub>
    </md-field-stub>
    <md-button-stub mdripple="true" type="submit" class="md-raised md-primary">
      Login
    </md-button-stub>
  </form>
  <router-link-stub to="/signup" tag="a" event="click" class="layout__link">
    Don't have an account?&nbsp;
    <span>
      Sign up
    </span></router-link-stub>
</div>
`;
